CMakeLists.txt

```diff
@@ -32,13 +32,25 @@ if (CMAKE_CXX_FLAGS_NATIVE STREQUAL "")
     endif()
 endif()
 
+option(ENABLE_CUDA "Enable CUDA support" OFF)
```

Comment:
Rename `ENABLE_CUDA` to `OPENSN_WITH_CUDA`. The prefix is important when OpenSn is used as a library, to prevent name collisions with other CMAKE variables defined by other projects. There is no namespacing for variables in cmake, so projects have to add the prefix themselves...

And we are using `WITH`. For example, see below `OPENSN_WITH_LUA`.